### PR TITLE
Updates for Raja & Umpire

### PR DESCRIPTION
**Pull request by gzagaris**
# Summary

- Updates Axom to RAJA-v0.12.1 and Umpire-v4.0.1

## Task List

- [x] Apply RAJA namespace changes
- [x] Use RAJA stable sort instead of cub sort
- [x] Make code backwards compatible with older versions of RAJA
- [x] Fix issue due to the multi-gpu capapbilities added in Umpire. We shouldn't call `umpire::ResourceManager` outside of main, especially when compiled with CUDA, since, the ResourceManager makes CUDA calls.
- [x] Added `+examples` variant to the umpire package to disable building the Umpire examples by default. This circumvents Umpire build failures when building with Spack.
- [x] Update RAJA TPL
- [x] Update Umpire TPL (no API changes)
- [x] Remove `axom::DEFAULT_ALLOCATOR_ID`, use `axom::getDefaultAllocatorID()` instead.
- [x] Resolve issue with hanging tests on `rzansel`
       - python was hanging b/c my environment was configured to use the python installed for another project.
- [x] Update rzgenie host-configs
- [x] Update rzansel host-configs
- [x] Update rzmanta host-configs
- [x] Resolve BVH test failure on `rzmanta` and `rzansel` using the IBM XL compiler
       - Appears to be a compiler bug. Resolved by writing that part of the code w/out lambdas.
- [x] Fix Memory error with CUDA
- [x] Remove AXOM_ENABLE_CUB option from host-config (and generator)
- [x] Add issue to update RAJA & Umpire docker images
- [x] Run clang-format
- [x] Update Release notes

### CODE REVIEW

File: src/axom/core/memory_management.hpp [hunk 1]
```diff
@@ -21,11 +21,8 @@
 
 namespace axom
 {
-#ifdef AXOM_USE_UMPIRE
-const int DEFAULT_ALLOCATOR_ID =
-  umpire::ResourceManager::getInstance().getAllocator("HOST").getId();
```
Comment (davidbeckingsale) [inline src/axom/core/memory_management.hpp:26]: 🎉

Comment (gzagaris) [reply src/axom/core/memory_management.hpp:26]: Thanks again for your help @davidbeckingsale

**Comment by gzagaris** [comment]
This should be ready to go. The branch is passing all of our specs on LC.

**Review by rhornung67 — APPROVED** [review]
Looks good. Thanks!

**Review by agcapps — APPROVED** [review]
Thank you for this work, @gzagaris .

**Review by davidbeckingsale — APPROVED** [review]

**Review by nselliott — APPROVED** [review]

**Pull request merged**